sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/subscribe/SubscribeProjectController.groovy

```diff
@@ -0,0 +1,37 @@
+package life.qbic.portal.sampletracking.components.projectoverview.subscribe
+
+import life.qbic.business.project.subscribe.SubscribeProject
+import life.qbic.business.project.subscribe.SubscribeProjectInput
+import life.qbic.business.project.subscribe.Subscriber
+
+/**
+ * <b>Controller allowing the view to call the subscribe project use case</b>
+ *
+ * <p>Used via the view to start the subscribe project use case with the currently selected project code and a provided subscriber. </p>
+ *
+ * @since 1.0.0
+ */
+class SubscribeProjectController {
+
+    SubscribeProjectInput subscribeProject
+
+    SubscribeProjectController(SubscribeProject subscribeProject) {
+        this.subscribeProject = subscribeProject
+    }
+    
+    /**
+     * Triggers the Subscribe Project case. If no project code or subscriber is provided, throws an {@link IllegalArgumentException}
+     * @param projectCode the code of the selected project
+     * @param projectCode the user that will be subscribed to the project
+     * @throws IllegalArgumentException in case the project code or subscriber is not provided
+     */
+    void subscribeProject(Subscriber subscriber, String projectCode) throws IllegalArgumentException{
+        if (! subscriber) {
+            throw new IllegalArgumentException("No subscriber provided for subscription.")
+        }
+        else if (! projectCode) {
+            throw new IllegalArgumentException("No project selected for subscription.")
+        }
```

Comment:
I would leave that to the use case since we have our input validator there. Check out the implementation with functional interfaces and let me know what you think. Double-checking is not really necessary at this point I think.
But I like that you considered that the arguments might be empty and that you used the groovy build-in to check for them.
```suggestion

```